GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,68 @@
+﻿namespace GenericsHomework
+{
+    public class Node<TValue>
+        where TValue : notnull
+    {
+        private TValue Value { get; set; }
+        public Node<TValue> Next { get; private set; }
+
+        public Node(TValue value)
+        {
+            Value = value;
+            Next = this;
+        }
+
+        public void Clear()
+        {
+            this.Next = this;
+        }//end of Clear method
+
+        public void Append(TValue value)
+        {
+            if (Exists(value))
+            {
+                throw new ArgumentException("Value exist in the list.");
+            }
+            else
+            {
+                Node <TValue> nn = new(value);
+                Node<TValue> cur = Next;
+
+                while(cur.Next != this)
+                {
+                    cur = cur.Next;
+                }
+
+                cur.Next = nn;
+                nn.Next = this;
```

Comment:
This works, but you may be able to simplify it by doing the following: 
```suggestion
            if (Exists(value))
            {
                throw new ArgumentException("Value exist in the list.");
            }
            el
            Node<T> newNode = new(value);
            newNode.Next = Next;
            Next = newNode;
            return newNode;
```